### PR TITLE
Check the spring messaging migration guide

### PR DESCRIPTION
**Pull request by Netyyyy**
As title.link the [issue](https://github.com/Azure/azure-sdk-for-java/issues/27478)

### CODE REVIEW

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 1]
```diff
@@ -1429,7 +1431,8 @@ IMPORTANT: The binder type is combined from `servicebus-queue` and `servicebus-t
 |*spring.cloud.stream.servicebus.topic*.bindings.<binding-name>.consumer.*checkpoint-mode* |*spring.cloud.stream.servicebus*.bindings.<binding-name>.consumer.*auto-complete*
 |===
 
-NOTE: The concurrency property will replace maxConcurrentSessions when sessionsEnabled is `true` and maxConcurrentCalls when sessionsEnabled is `false`. Enabling auto-complete is equal to `RECORD` checkpoint mode, and oppositely the `MANUAL` mode.
+NOTE: The concurrency property will be replaced by the maxConcurrentSessions when sessionsEnabled is `true` and the maxConcurrentCalls when sessionsEnabled is `false`.
+NOTE: Enabling auto-complete is equal to `RECORD` checkpoint mode, and oppositely the `MANUAL` mode.
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1435]: This format is not correct, 
image
we expect the rendering effects before this change

**Review by saragluna — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 2]
```diff
@@ -931,22 +931,22 @@ the prefix remains to be as `spring.cloud.azure.servicebus`.
...
-* Drop message header of `AzureHeaders.RAW_ID`. Please use `ServiceBusMessageHeaders.MESSAGE_ID` instead.
+* Drop `ServiceBusQueueInboundChannelAdapter` and `ServiceBusTopicInboundChannelAdapter` with the listening to a Service Bus queue/topic entity function moved to `ServiceBusInboundChannelAdapter`.
+* Divide the constructor from `ServiceBusXXXInboundChannelAdapter(String, SubscribeByGroupOperation, String)` into `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer)` and `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer, ListenerMode)`.
+* Drop API `ServiceBusQueueOperation/ServiceBusTopicOperation#setCheckpointConfig`. To set the checkpoint configuration for the inbound channel adapter, users can call the method `ServiceBusContainerProperties#setAutoComplete` instead. To disable the auto-complete mode is equivalent to `MANUAL` checkpoint mode and to enable it will trigger the `RECORD` mode.
+* Drop API `ServiceBusQueueOperation/ServiceBusTopicOperation#setClientConfig` . To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
```
[the remark this answers is not in the document]
Comment (yiliuTo) [reply docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:946]: Hi @saragluna , this has been recorded in the first item of api changes, here to specifically mention the APIs is to suggest users how to find the replacement.

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 3]
```diff
@@ -931,22 +931,22 @@ the prefix remains to be as `spring.cloud.azure.servicebus`.
 |*spring.cloud.azure*.resource-group|*spring.cloud.azure.servicebus*.resource.resource-group
 |*spring.cloud.azure.servicebus*.transport-type |*spring.cloud.azure.servicebus*.client.transport-type
 |*spring.cloud.azure.servicebus*.retry-options.retry-mode |*spring.cloud.azure.servicebus*.retry.mode
-|*spring.cloud.azure.servicebus*.retry-options.max-retries |*spring.cloud.azure.servicebus*.retry.exponential.max-retries or *spring.cloud.azure.servicebus*.retry.fixed.max-retries, should be configured according to *spring.cloud.azure.servicebus*.retry.mode
-|*spring.cloud.azure.servicebus*.retry-options.delay |*spring.cloud.azure.servicebus*.retry.exponential.base-delay or *spring.cloud.azure.servicebus*.retry.fixed.delay, should be configured according to *spring.cloud.azure.servicebus*.retry.mode
+|*spring.cloud.azure.servicebus*.retry-options.max-retries |*spring.cloud.azure.servicebus*.retry.exponential.max-retries or *spring.cloud.azure.servicebus*.retry.fixed.max-retries, should be configured depends on *spring.cloud.azure.servicebus*.retry.mode=*fixed* or *exponential*
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:934]: we should use "according to" or "depending on"

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 4]
```diff
@@ -931,22 +931,22 @@ the prefix remains to be as `spring.cloud.azure.servicebus`.
...
+|*spring.cloud.azure.servicebus*.retry-options.delay |*spring.cloud.azure.servicebus*.retry.exponential.base-delay or *spring.cloud.azure.servicebus*.retry.fixed.delay, should be configured depends on *spring.cloud.azure.servicebus*.retry.mode=*fixed* or *exponential*
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:935]: same here

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 5]
```diff
@@ -931,22 +931,22 @@ the prefix remains to be as `spring.cloud.azure.servicebus`.
...
-* Drop API `setClientConfig` of `ServiceBusQueueOperation` and `ServiceBusTopicOperation`. To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
-* Drop `CompletableFuture` support of `ServiceBusTemplate` and `DefaultMessageHandler` and support Reactor instead.
-* Add new API of `setDefaultEntityType` for `ServiceBusTemplate` to specify the entity type, which is required when no bean of `PropertiesSupplier&lt;String, ProducerProperties&gt;` is provided for the `ProducerProperties#entityType`.
-* Drop message header of `AzureHeaders.RAW_ID`. Please use `ServiceBusMessageHeaders.MESSAGE_ID` instead.
+* Drop `ServiceBusQueueInboundChannelAdapter` and `ServiceBusTopicInboundChannelAdapter` with the listening to a Service Bus queue/topic entity function moved to `ServiceBusInboundChannelAdapter`.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:943]: “with the listening to a Service Bus queue/topic entity function moved to `ServiceBusInboundChannelAdapter`”  -> ", and move the functionality to listen to a Service Bus queue/topic entity to `ServiceBusInboundChannelAdapter`"
let's try to use simple sentences.

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 6]
```diff
@@ -931,22 +931,22 @@ the prefix remains to be as `spring.cloud.azure.servicebus`.
...
+* Divide the constructor from `ServiceBusXXXInboundChannelAdapter(String, SubscribeByGroupOperation, String)` into `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer)` and `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer, ListenerMode)`.
+* Drop API `ServiceBusQueueOperation/ServiceBusTopicOperation#setCheckpointConfig`. To set the checkpoint configuration for the inbound channel adapter, users can call the method `ServiceBusContainerProperties#setAutoComplete` instead. To disable the auto-complete mode is equivalent to `MANUAL` checkpoint mode and to enable it will trigger the `RECORD` mode.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:945]: I don't like the way to use `\` in API reference, looks like a bit unformal, we could use "Drop APIs `ServiceBusQueueOperation#setCheckpointConfig` and `ServiceBusTopicOperation#setCheckpointConfig`"

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc (hunk 2 above)
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:946]: same here

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 7]
```diff
@@ -1100,13 +1100,13 @@ version 2 of link:https://search.maven.org/artifact/com.azure.spring/azure-sprin
...
-* Drop API `setClientConfig` of `ServiceBusQueueOperation` and `ServiceBusTopicOperation`. To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
-* Drop `CompletableFuture` support of `ServiceBusTemplate` and `DefaultMessageHandler` and support Reactor instead.
-* Add new API of `setDefaultEntityType` for `ServiceBusTemplate` to specify the entity type, which is required when no bean of `PropertiesSupplier&lt;String, ProducerProperties&gt;` is provided for the `ProducerProperties#entityType`.
-* Drop message header of `AzureHeaders.RAW_ID`. Please use `ServiceBusMessageHeaders.MESSAGE_ID` instead.
+* Drop `ServiceBusQueueInboundChannelAdapter` and `ServiceBusTopicInboundChannelAdapter` with the listening to a Service Bus queue/topic entity function moved to `ServiceBusInboundChannelAdapter`.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1103]: same here

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 8]
```diff
@@ -1100,13 +1100,13 @@ version 2 of link:https://search.maven.org/artifact/com.azure.spring/azure-sprin
...
+* Divide the constructor from `ServiceBusXXXInboundChannelAdapter(String, SubscribeByGroupOperation, String)` into `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer)` and `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer, ListenerMode)`.
+* Drop API `ServiceBusQueueOperation/ServiceBusTopicOperation#setCheckpointConfig`. To set the checkpoint configuration for the inbound channel adapter, users can call the method `ServiceBusContainerProperties#setAutoComplete` instead. To disable the auto-complete mode is equivalent to `MANUAL` checkpoint mode and to enable it will trigger the `RECORD` mode.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1105]: same here

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 9]
```diff
@@ -1100,13 +1100,13 @@ version 2 of link:https://search.maven.org/artifact/com.azure.spring/azure-sprin
...
+* Drop API `ServiceBusQueueOperation/ServiceBusTopicOperation#setClientConfig` . To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1106]: same here

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 10]
```diff
@@ -1417,8 +1419,8 @@ IMPORTANT: The binder type is combined from `servicebus-queue` and `servicebus-t
 |*spring.cloud.azure*.resource-group|*spring.cloud.azure.servicebus*.resource.resource-group
 |*spring.cloud.azure.servicebus*.transport-type |*spring.cloud.azure.servicebus*.client.transport-type
 |*spring.cloud.azure.servicebus*.retry-options.retry-mode |*spring.cloud.azure.servicebus*.retry.mode
-|*spring.cloud.azure.servicebus*.retry-options.max-retries |*spring.cloud.azure.servicebus*.retry.exponential.max-retries or *spring.cloud.azure.servicebus*.retry.fixed.max-retries, should be configured according to *spring.cloud.azure.servicebus*.retry.mode
-|*spring.cloud.azure.servicebus*.retry-options.delay |*spring.cloud.azure.servicebus*.retry.exponential.base-delay or *spring.cloud.azure.servicebus*.retry.fixed.delay, should be configured according to *spring.cloud.azure.servicebus*.retry.mode
+|*spring.cloud.azure.servicebus*.retry-options.max-retries |*spring.cloud.azure.servicebus*.retry.exponential.max-retries or *spring.cloud.azure.servicebus*.retry.fixed.max-retries, should be configured depends on *spring.cloud.azure.servicebus*.retry.mode=*fixed* or *exponential*
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1422]: same here

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 11]
```diff
@@ -1417,8 +1419,8 @@ IMPORTANT: The binder type is combined from `servicebus-queue` and `servicebus-t
...
+|*spring.cloud.azure.servicebus*.retry-options.delay |*spring.cloud.azure.servicebus*.retry.exponential.base-delay or *spring.cloud.azure.servicebus*.retry.fixed.delay, should be configured depends on *spring.cloud.azure.servicebus*.retry.mode=*fixed* or *exponential*
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1423]: same here

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 12]
```diff
@@ -931,22 +931,22 @@ the prefix remains to be as `spring.cloud.azure.servicebus`.
...
-* Drop API `setClientConfig` of `ServiceBusQueueOperation` and `ServiceBusTopicOperation`. To configure the underlying `ServiceBusProcessorClient` used by the inbound channel adapter, users can use `ServiceBusContainerProperties` instead.
-* Drop `CompletableFuture` support of `ServiceBusTemplate` and `DefaultMessageHandler` and support Reactor instead.
-* Add new API of `setDefaultEntityType` for `ServiceBusTemplate` to specify the entity type, which is required when no bean of `PropertiesSupplier&lt;String, ProducerProperties&gt;` is provided for the `ProducerProperties#entityType`.
-* Drop message header of `AzureHeaders.RAW_ID`. Please use `ServiceBusMessageHeaders.MESSAGE_ID` instead.
+* Drop `ServiceBusQueueInboundChannelAdapter` and `ServiceBusTopicInboundChannelAdapter` with the listening to a Service Bus queue/topic entity function moved to `ServiceBusInboundChannelAdapter`.
+* Divide the constructor from `ServiceBusXXXInboundChannelAdapter(String, SubscribeByGroupOperation, String)` into `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer)` and `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer, ListenerMode)`.
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:944]: how about we splitting this item to two:
* Change the constructor from `ServiceBusQueueInboundChannelAdapter(String, SubscribeByGroupOperation, String)` to `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer)` and `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer, ListenerMode)`
* Change the constructor from `ServiceBusTopicInboundChannelAdapter(String, SubscribeByGroupOperation, String)` to `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer)` and `ServiceBusInboundChannelAdapter(ServiceBusMessageListenerContainer, ListenerMode)`

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 13]
```diff
@@ -1100,13 +1100,13 @@ version 2 of link:https://search.maven.org/artifact/com.azure.spring/azure-sprin
...
```
Comment (yiliuTo) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1104]: same here

**Review by yiliuTo — CHANGES_REQUESTED** [review]

**Review by yiliuTo — APPROVED** [review]
LGTM

**Review by saragluna — APPROVED** [review]

**Pull request merged**